core/sdk-core/src/test/java/software/amazon/awssdk/core/exception/SdkExceptionMessageTest.java

```diff
@@ -47,4 +47,27 @@ public void noMessage_causeWithoutMessage_implies_noMessage() {
     public void noMessage_causeWithMessage_implies_messageFromCause() {
         assertThat(SdkException.builder().cause(new Exception("bar")).build().getMessage()).isEqualTo("bar");
     }
+
+    @Test
+    public void defaultAttemptCount_shouldReturnOne() {
+        assertThat(SdkException.builder().build().getAttempts()).isEqualTo(0);
+    }
+
+    @Test
+    public void getAttempts_WithExplicitAttemptCount_ReturnsOneBased() {
+        assertThat(SdkException.builder().message("foo").attemptCount(2).build().getAttempts()).isEqualTo(2);
+    }
```

Comment:
Consider using ParameterizedTest if we are verifying the same thing with different values.